app/helpers/pipeline_runs_helper.rb

```diff
@@ -181,4 +181,25 @@ def pipeline_version_has_coverage_viz(pipeline_version)
   def pipeline_version_at_least_2(pipeline_version)
     pipeline_version_at_least(pipeline_version, PIPELINE_VERSION_2)
   end
+
+  def check_for_user_error(failed_stage)
+    if failed_stage.step_number == 1
+      user_input_validation_file = "#{failed_stage.pipeline_run.host_filter_output_s3_path}/#{PipelineRun::INPUT_VALIDATION_NAME}"
+      invalid_step_input_file = "#{failed_stage.pipeline_run.host_filter_output_s3_path}/#{PipelineRun::INVALID_STEP_NAME}"
+      if file_generated_since_run(failed_stage, user_input_validation_file)
+        file = Tempfile.new
+        downloaded = PipelineRun.download_file_with_retries(user_input_validation_file, file.path, 3, false)
+        validation_error = downloaded ? JSON.parse(File.read(file))["Validation error"] : nil
+        file.unlink
+        return ["FAULTY_INPUT", validation_error] if validation_error
```

Comment:
hmmmm. It seems a bit long and non-specific. 
Right now the 2 codes are FAULTY_INPUT and INSUFFICIENT_READS. I'm worried INPUT_VALIDATION_ERROR might sound like it would also encompass INSUFFICIENT_READS. I think I'll leave it is as for now but we can always change in the future